docs/tutorials/pytorch-export.ipynb

```diff
@@ -125,22 +137,31 @@
       "source": [
         "### Export with dynamic batch dimension\n",
         "\n",
-        "_This is a new feature and will be included in pip installation after the 2.4 release, or if using `torch_xla` nightly. Once PyTorch/XLA 2.4 is released, this will be converted into a running example. Using the nightly `torch` and `torch_xla` will likely lead to notebook failures in the meantime._\n",
         "\n",
-        "Dynamic batch dimensions can be specified as a part of the initial `torch.export` step. The FX Graph's symint information is used to export to dynamic StableHLO.\n",
+        "Dynamic batch dimensions can be specified as a part of the initial `torch.export` step. The FX graph's SymInt information is used to export to dynamic StableHLO.\n",
         "\n",
-        "In this example, we specify that dim 0 of the sample input is dynamic, which propagates shape using a `tensor<?x3x224x224xf32>`.\n",
+        "In this example, we specify that dimension 0 of the sample input is dynamic, which propagates shape using a `tensor<?x3x224x224xf32>`."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# TODO: FIX\n",
+        "# Kernel crashes with \"INVALID_ARGUMENT: Non-broadcast dimensions must not be dynamic\"\n",
```

Comment:
TorchXLA coverage of dynamic shapes isn't great, for dynamic shape / export workflows we actually recommend torch_xla2 which uses PT->JAX->StableHLO, and has much better coverage for dynamic shapes:
https://github.com/pytorch/xla/tree/master/experimental/torch_xla2

Perhaps we just make a note of "For dynamic shape export we recommend using torch_xla2 (link), which converts PyTorch models to JAX prior to exporting to StableHLO and has high opset coverage along with great dynamic shape support."

If we want to add a tutorial on using torch_xla2, it likely wouldn't be too hard, here's a recent example ([discord invite needed to view](https://discord.gg/ZKXq7b3V8A)):
https://discord.com/channels/999073994483433573/1281512734902714379/1298361762839527467

I believe the following should work:

```
# Install torch_xla2
pip install torch-xla2 # Note: may need to follow GH instructions linked above but pypi package looks up to date.

# Export the program from native pytorch using torch.export
from torch.export import Dim
batch = Dim(\"batch\", min=4, max=6)
dynamic_shapes = ({0: batch},)
dynamic_export = export(resnet18, sample_input, dynamic_shapes=dynamic_shapes)

# Convert the exported program to StableHLO using torch_xla2
from torch_xla2 import export
stablehlo = export.exported_program_to_stablehlo(exported).mlir_module()
print(stablehlo)
```

Note the print will include _lots_ of debug info, so maybe use some indexing, or print starting with the word `@main` for 1k characters or something to keep the tutorial readable. I.e. `print(stablehlo[1000:2000])` or something.

Reply:
Thanks for the notes, I'll include it in the PR.

Reply:
@GleasonK - I just updated this section with a Tip about `torch_xla2` to unblock merging. We can update the page iteratively based on the following.

The `torch_xla2` code was still failing with:

```python-traceback
RuntimeError: ('No lowering found for\n\nWhile executing %_native_batch_norm_legit_functional : 
...
```
I can share the full traceback here or in a separate issue if you're interested.

Alternatively, I can use a different, simpler example (like on Discord) just for the dynamic shapes section. For this, we can have a separate `torch_xla2` section at the end.
---

Aside, I checked with the team internally, and they identified this as a potential `MaxPoolNd` lowering bug. Victor was able to get dynamic shapes export working with some explicit `maxpool` and `avgpool` settings: [Kaggle notebook with working snippet](https://www.kaggle.com/code/pavithraes/export-pytorch-to-stablehlo-with-dynamic-shapes). 

If the dynamic shapes support is not-stable, I think it's best to exclude code samples. That said, I'm happy to include the working example if you'd like to have it. :)

Reply:
Let's use the note for now, repro would be great and can update in the near future. This should be stable :) but if it isn't then lets avoid the code example as you said.

Reply:
Could you open repro as a bug ticket on the ptxla repo?
https://github.com/pytorch/xla/tree/master